### DISCUSSION TITLE
Generate vendor configs for devices without launching virtual instances

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#1330] [Ideas]
Some devices are difficult to prepare and spin up; it would be nice to be able to generate configs for various vendor nodes, without necessarily having access to virtual instances for each.

One could build a virtual topology using FRR nodes, while having a physical network made of Cisco devices. Without changing the actual network configuration, generating candidate configs for the physical nodes would be useful

**Comment by ipspace** [comment]
**netlab create** followed by **netlab initial -o** (https://netlab.tools/netlab/initial/#usage) should do the trick.

**Comment by sdargoeuves** [comment]
> One could build a virtual topology using FRR nodes, while having a physical network made of Cisco devices. Without changing the actual network configuration, generating candidate configs for the physical nodes would be useful

That's an interesting idea, and it means you don't even need the image of the Cisco device (in your example) in netlab. Great for a migration to a vendor you do not know, you can check the candidate configuration.
